packages/wpcom-checkout/src/transformations.ts

```diff
@@ -443,6 +444,11 @@ function canDisplayIntroductoryOfferDiscountForProduct( product: ResponseCartPro
 	if ( isJetpackSocialAdvancedSlug( product.product_slug ) ) {
 		return false;
 	}
+
+	if ( isDomainProduct( product ) ) {
```

Comment:
Is this true of _all_ domain product introductory offers?

Reply:
Also, I think we could add a comment here to explain why domain products shouldn't display the introductory offer discount. It could be the same as the other one below ("Introductory offers for Domains are used for premium domains and should not be considered as a discount.")

Reply:
> Is this true of _all_ domain product introductory offers?

Thanks to you both @sirbrillig and @leonardost for the feedback!

I added a method `isDomainIntroductoryOfferVisibleAsDiscount` to clarify what is happening. Also, if some domain products use those introductory offers differently in the future, we can change that method without changing the checkout code itself.